apis/Google.Cloud.AspNetCore.DataProtection.SecretManager/Google.Cloud.AspNetCore.DataProtection.SecretManager/GoogleCloudDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.SecretManager.V1;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Google.Cloud.AspNetCore.DataProtection.SecretManager;
+
+/// <summary>
+/// Static class to provide extension methods for modifying a <see cref="IDataProtectionBuilder"/> to work with Google Cloud Secret Manager.
+/// </summary>
+public static class GoogleCloudDataProtectionBuilderExtensions
+{
+    /// <summary>
+    /// Configures data protection system to persist keys in Google Cloud Secret Manager.
+    /// </summary>
+    /// <param name="builder">The data protection builder to configure. Must not be null.</param>
+    /// <param name="projectName">The project id in which the keys are stored. Must not be null</param>
```

Comment:
Nit: Can we just name it as projectId so that name and description is consistent? (here and in other relevant places)

Reply:
done